frontend/packages/ceph-storage-plugin/src/components/block-pool/create-block-pool.tsx

```diff
@@ -0,0 +1,171 @@
+import { useTranslation } from 'react-i18next';
+
+import { referenceForModel, K8sResourceKind } from '@console/internal/module/k8s';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { useDeepCompareMemoize } from '@console/shared';
+import {
+  BreadCrumbs,
+  LoadingInline,
+  history,
+  resourcePathFromModel,
+} from '@console/internal/components/utils';
+import {
+  ClusterServiceVersionModel,
+  ClusterServiceVersionKind,
+} from '@console/operator-lifecycle-manager';
+import { Button } from '@patternfly/react-core';
+import { k8sCreate } from '@console/internal/module/k8s/resource';
+import { Modal } from '@console/shared/src/components/modal';
+
+import { PoolBody } from './body';
+import { PoolFooter } from './footer';
+import { CephClusterKind, StoragePoolKind } from '../../types';
+import { cephClusterResource } from '../../resources';
+import { CEPH_EXTERNAL_CR_NAME } from '../../constants';
+import { CephBlockPoolModel } from '../../models';
+import {
+  blockPoolReducer,
+  blockPoolInitialState,
+  BlockPoolActionType,
+  getPoolKindObj,
+  getErrorMessage,
+} from '../../utils/block-pool';
+
+import './create-block-pool.scss';
+
+const CreateBlockPool: React.FC<CreateBlockPoolProps> = ({ match }) => {
+  const {
+    params: { ns, appName },
+    url,
+  } = match;
+  const { t } = useTranslation();
+
+  const csvResource = {
+    kind: referenceForModel(ClusterServiceVersionModel),
+    name: appName,
+    namespace: ns,
+    isList: false,
+  };
+
+  const [state, dispatch] = React.useReducer(blockPoolReducer, blockPoolInitialState);
+  const [csv] = useK8sWatchResource<ClusterServiceVersionKind>(csvResource);
+  const [cephClusters, isLoaded, loadError] = useK8sWatchResource<CephClusterKind[]>(
+    cephClusterResource,
+  );
+
+  const memoizedCSV: K8sResourceKind = useDeepCompareMemoize(csv, true);
+  const cephCluster: CephClusterKind = useDeepCompareMemoize(cephClusters[0], true);
+
+  const storageClusterListPage = `${resourcePathFromModel(
+    ClusterServiceVersionModel,
+    appName,
+    ns,
+  )}/${referenceForModel(CephBlockPoolModel)}`;
+
+  const onClose = () => {
+    history.push(storageClusterListPage);
+  };
+
+  // Create new pool
+  React.useEffect(() => {
+    if (state.isSubmitted) {
+      dispatch({ type: BlockPoolActionType.SET_IS_SUBMITTED, payload: false });
+      const poolObj: StoragePoolKind = getPoolKindObj(state);
+
+      dispatch({ type: BlockPoolActionType.SET_INPROGRESS, payload: true });
+      k8sCreate(CephBlockPoolModel, poolObj)
+        .then(() => history.push(`${storageClusterListPage}/${state.poolName}`))
+        .finally(() => dispatch({ type: BlockPoolActionType.SET_INPROGRESS, payload: false }))
+        .catch((err) =>
+          dispatch({
+            type: BlockPoolActionType.SET_ERROR_MESSAGE,
+            payload: getErrorMessage(err.message) || 'Could not create block pool.',
+          }),
+        );
```

Comment:
This logic should be in onSubmit handler.

Reply:
I have used footer as a different component to re-use, so we can't place this indie Onsubmit handler

Reply:
Or you can pass a handler as a prop to the footer component

Reply:
ack